### PR TITLE
docker-compose files and yarn dependency cleanup

### PR DESCRIPTION
**Pull request by c0m3d1an**
Changes to be committed:
	new file:   Dockerfile
	new file:   docker-compose.yml
	modified:   package.json
	modified:   yarn.lock

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,15 +42,18 @@
     "core-js": "^2.6.1",
     "hammerjs": "^2.0.8",
     "idb-keyval": "^3.2.0",
+    "jquery": "^1.9.1",
```
Comment (dcaslin) [inline package.json]: What's with the package.json changes besides the docker start?

**Comment by dcaslin** [comment]
Dockerfile stuff all looks great! Can you either: 

1) Restore the package.json and yarn.lock back to match the origin

2) I can just grab this Dockerfile and paste the docker run in package.json into the latest

**Comment by c0m3d1an** [comment]
My assumption is that you started this project using npm and switched (at some point) to yarn.  When I tried to build your site using yarn I got multiple errors for packages that would not build because their dependencies were not installed.  I updated the yarn and package.json files to include those dependencies so ng serve would spin up correctly.

Additionally, since Bungo REQUIRES the site to be SSL (the return URL MUST start with https://) I needed a line in there for a custom start command with the --ssl switch(which creates dummy certs if they don't exist).  I don't believe the jquery include nor the correction of the line with the tab at the end is required, but the line with "docker-start": "ng serve --host 0.0.0.0 --ssl", would be EXTREMELY helpful to maintain.

As for yarn.lock, your program WILL NOT BUILD on any recent linux I tried building it on due to version and dependency requirements, so the yarn.lock changes (were)are absolutely required for this program to build successfully.  Due to the age of angular you're using and the age of the build dependencies, if the versions don't exist in the yarn.lock, it tries to pull the most recent versions and fails building the components that depend on them.

I see your commit and conflict request only require me pulling edits from package.json, and you'll paste them back in there.  If that IS the only requirement, then I can re-sub with the Dockerfile, docker-compose.yml, and yarn.lock.

Make sense?  LMK.

**Comment by dcaslin** [comment]
I'm so sorry, I somehow missed your response! I'll read this in more detail tomorrow and get back. In the meantime: 

- You are correct that I use yarn as my package manager (I build on Mac and Windows 10 for personal dev)

- You are correct about HTTPS. That -ssl flag is really nice! What I do now is just hack it for local dev where I run my webpack server on http and have to manually delete the http**s** from the URL on the Bungie sign in call.

**Comment by dcaslin** [comment]
> I see your commit and conflict request only require me pulling edits from package.json, and you'll paste them back in there. If that IS the only requirement, then I can re-sub with the Dockerfile, docker-compose.yml, and yarn.lock.

This would be perfect! Please go ahead with that, thanks!

**Comment by c0m3d1an** [comment]
I did actually re-submit the branch as discussed after my comments above.  :-)  It still works without the deps in the package.json, although it does compile with dependency errors.  (Although still completely fails without the deps in the yarn.lock).

**Comment by dcaslin** [comment]
I figured out the jquery mystery, bootstrap (which I'm using for pure CSS and no JS) had it as a dependency. Just merged, thanks!

**Comment by c0m3d1an** [comment]
Oh awesome, good to hear!  I am NOT remotely a node or angular expert, so I was pretty much just flailing blindly in an attempt to get the docker code working.  :-)

**Comment by dcaslin** [comment]
Np! I just pushed 7.8.0 after adding back in the docker-start command. If you get a chance give it a run and see if it actually works (hopefully I didn't break anything)

**Comment by c0m3d1an** [comment]
Works perfectly!  Yay!

**Pull request merged**